code/species_defs/species_defs.cpp

```diff
@@ -328,6 +328,34 @@ void parse_species_tbl(const char *filename)
 			if (optional_string("$Countermeasure type:"))
 				stuff_string(species->cmeasure_name, F_NAME, NAME_LENGTH);
 
+			if (optional_string("$Custom wingmate formations:")) {
+
+				while (check_for_string("(")) {
+					SCP_vector<vec3d> current_formation;
+
+					if (stuff_vec3d_list(current_formation) != MAX_SHIPS_PER_WING - 1) {
+						Warning(LOCATION, "A custom formation for species %s did not have 5 positions. Ignoring.", species->species_name);
+					} else {
+						species->formations.emplace_back();
+						std::copy_n(current_formation.begin(), MAX_SHIPS_PER_WING-1, species->formations.back().begin());
```

Comment:
You can use `push_back(std::move(current_formation))` here.  This will avoid the copying, as it simply makes the new vector point to the old vector's data.

Reply:
Well, it isn't actually a vector of vectors, but a vector of arrays so I don't think you can do this without a copy.

Reply:
Oh, you are correct. :thonk:

By the way, that 5 in the `Warning` should also be the SHIPS-1 number.  Use SIZE_T_ARG, e.g.
```
error_display(0, "String too long.  Length = " SIZE_T_ARG ".  Max is %i.\n", len, max);
```